c11threads.h

```diff
@@ -159,14 +170,15 @@ static inline int mtx_trylock(mtx_t *mtx)
 
 static inline int mtx_timedlock(mtx_t *mtx, const struct timespec *ts)
 {
-	int res;
+	int res = 0;
 #ifdef C11THREADS_NO_TIMED_MUTEX
 	/* fake a timedlock by polling trylock in a loop and waiting for a bit */
-	struct timeval now;
-	struct timespec sleeptime;
+	struct timeval now = {0};
+	struct timespec sleeptime = {0};
 
 	sleeptime.tv_sec = 0;
 	sleeptime.tv_nsec = C11THREADS_TIMEDLOCK_POLL_INTERVAL;
+	nanosleep(&sleeptime, NULL);
```

Comment:
Add a comment explaining why we need a sleep at this point.

Reply:
I will do more tests to understand why. It shouldnt need it...

Reply:
That would be ideal if you have the time to look into it, but even if you can't figure it out, at least add a comment saying why you added that nanosleep and how it breaks without it, to avoid confusion later.

Reply:
I found the problem: missing `mtx_init` in `test.c` =)